File: bin/Data/AUI/Periodic/Sources/code_acolorwidget.cpp
// AColorWidget application source code
#include <Urho3D/AUI/AColorWidget.h>
#include "PeriodicApp.h"

void PeriodicApp::setup_acolorwidget( AWidget *layout )
{
    PODVector<AWidget*> dest;
    layout->SearchWidgetClass( "TBButton", dest );
    for (unsigned ii = 0; ii < dest.Size(); ii++)
        SubscribeToEvent(dest[ii], E_WIDGETEVENT, URHO3D_HANDLER(PeriodicApp, HandleAcolorwidgetEvent ));
}

void PeriodicApp::HandleAcolorwidgetEvent(StringHash eventType, VariantMap& eventData)
{
    using namespace WidgetEvent;
    AWidget* widget = static_cast<AWidget*>(eventData[P_TARGET].GetPtr());
    if ( widget == NULL ) return;
    if (eventData[P_TYPE] == UI_EVENT_TYPE_CLICK)
    {
        if (widget->GetId() == "uicolorwidgetcode" )
        {
            AppLog( "AColorWidget support : " + widget->GetId() + " was pressed " );
            ViewCode ( "Sources/code_acolorwidget.cpp", widget->GetParent() );
        }
        if (widget->GetId() == "uicolorwidgetlayout" )
        {
            AppLog( "AColorWidget support : " + widget->GetId() + " was pressed ");
            ViewCode ( "Scenes/layout_acolorwidget.ui.txt", widget->GetParent() );
        }

        if (widget->GetId() == "uicolorwidgetred" )
        {
            AppLog( "AColorWidget action : " + widget->GetId() + " was pressed ");
            AColorWidget *clw = static_cast<AColorWidget*>(widget->FindWidget("colorwidgetdemo"));
            if (clw)
                clw->SetColorString("#FF1111");
        }
        if (widget->GetId() == "uicolorwidgetgreen" )
        {
            AppLog( "AColorWidget action : " + widget->GetId() + " was pressed ");
            AColorWidget *clw = static_cast<AColorWidget*>(widget->FindWidget("colorwidgetdemo"));
            if (clw)
                clw->SetColorString("#11FF11");
        }
        if (widget->GetId() == "uicolorwidgetblue" )
        {
            AppLog( "AColorWidget action : " + widget->GetId() + " was pressed ");
            AColorWidget *clw = static_cast<AColorWidget*>(widget->FindWidget("colorwidgetdemo"));
            if (clw)
                clw->SetColorString("#1111FF");
        }
    }
}

File: bin/Data/AUI/Periodic/LuaScripts/code_acolorwidget.lua
-- AColorWidget application source code

function setup_uicolorwidget( layout )
    local dest = layout:SearchAWidgetClass( "TBButton" )
    for i, awid in ipairs(dest) do
        SubscribeToEvent(awid, "WidgetEvent", "HandleUicolorwidgetEvent")
	end
end

function HandleUicolorwidgetEvent( eventType, eventData )
    local widget = eventData["Target"]:GetPtr("AWidget")
    if widget == nil then
 		return;
	end
    if eventData["Type"]:GetInt() == UI_EVENT_TYPE_CLICK then
        if widget:GetId() == "uicolorwidgetcode" then
            AppLog( "AColorWidget support : " .. widget:GetId() .. " was pressed " )
            ViewCode ( "LuaScripts/code_acolorwidget.lua", widget:GetParent())
        end
        if widget:GetId() == "uicolorwidgetlayout" then
            AppLog( "AColorWidget support : " .. widget:GetId() .. " was pressed ")
            ViewCode ( "Scenes/layout_acolorwidget.ui.txt", widget:GetParent())
        end
        if widget:GetId() == "uicolorwidgetred" then
            AppLog( "AColorWidget action : " .. widget:GetId() .. " was pressed ")
            local clw = widget:FindWidget("colorwidgetdemo")
            if clw ~= nil then
            	local cw1 = tolua.cast(clw, "AColorWidget")
                cw1:SetColorString("#FF1111")
			end
        end
        if widget:GetId() == "uicolorwidgetgreen" then
            AppLog( "AColorWidget action : " .. widget:GetId() .. " was pressed ")
            local clw = widget:FindWidget("colorwidgetdemo")
            if clw ~= nil then
            	local cw1 = tolua.cast(clw, "AColorWidget")
                cw1:SetColorString("#11FF11")
			end
        end
        if widget:GetId() == "uicolorwidgetblue" then
            AppLog( "AColorWidget action : " .. widget:GetId() .. " was pressed ")
            local clw = widget:FindWidget("colorwidgetdemo")
            if clw ~= nil then
            	local cw1 = tolua.cast(clw, "AColorWidget")
                cw1:SetColorString("#1111FF")
			end
        end
    end
end

